Source/Core/Common/Thread.cpp

```diff
@@ -119,19 +121,30 @@ void SetThreadAffinity(std::thread::native_handle_type thread, u32 mask)
 {
 #ifdef __APPLE__
   thread_policy_set(pthread_mach_thread_np(thread), THREAD_AFFINITY_POLICY, (integer_t*)&mask, 1);
-#elif (defined __linux__ || defined BSD4_4 || defined __FreeBSD__) && !(defined ANDROID)
+#elif (defined __linux__ || defined BSD4_4 || defined __FreeBSD__ || defined __NetBSD__) && !(defined ANDROID)
 #ifdef __FreeBSD__
   cpuset_t cpu_set;
+#elif defined(__NetBSD__)
+  cpuset_t *cpu_set = cpuset_create();
 #else
   cpu_set_t cpu_set;
 #endif
+#ifndef __NetBSD__
   CPU_ZERO(&cpu_set);
+#endif
 
   for (int i = 0; i != sizeof(mask) * 8; ++i)
     if ((mask >> i) & 1)
+#ifndef __NetBSD__
```

Comment:
Please make two ifndef blocks. This looks horribly error prone. The difference in indentation looked like a style issue at first, until I realised the cpuset_set was guarded by an if statement...